Dev -> stable upon new release (1.0.5 is stable imo)

karllindmark opened the pull request:
@peter-budo Suitable to merge dev into stable now that 1.0.5 has been done?

peter-budo committed:
Update of super pom dependency, updated ABS version made Lint goal available to Maven

peter-budo committed:
- Parent pom updating version of reporting pom
- Instrumentation pom updating spoon library version and made version a property for easier maintenance since use in two places
- LoginHtmlParserTest, trying to address the issue of JUnit test execution from IDE, Maven command line is working fine

peter-budo committed:
Merge branch 'dev' into instrumentation

Merge instrumentation with dev branch

peter-budo committed:
Added Windows files to .gitignore exclude. ixed JUnit tests with local htm files. Removed irrelevant log file from repo

Commit:
Merge pull request #36 from peter-budo/instrumentation

IDE testing imroved

Commit:
Addressing the login bug (seems to be related to a timeout) as well as enabling the user to save his password

Commit:
Updating the version numbers for release.

peter-budo committed:
Updated parent pom so we are using android maven plugin 3.6.0 to avoid issue with symlinks

Commit:
Merge pull request #37 from peter-budo/dev

Android SDK update and Maven plugin issue

Commit:
Merge branch 'dev' of https://github.com/ninetwozero/Battle-Chat into dev

Commit:
Addressing the issue where the attributes tied to the chat have been changed.

Commit:
Updating the version numbers

Commit:
Addressing the issues with the loading of messages

Commit:
Oops

Commit:
Merge pull request #38 from ninetwozero/1.0.5

1.0.5 -> dev

peter-budo commented on app/src/com/ninetwozero/battlechat/BattleChat.java:
```diff
@@ -76,7 +76,8 @@ public static void saveToSharedPreferences(Context c) {
 
 			editor.putLong(Keys.Session.USER_ID, mSession.getUser().getId());
-			editor.putString(Keys.Session.USERNAME, mSession.getUser().getUsername());
+            editor.putString(Keys.Session.USERNAME, mSession.getUser().getUsername());
```
Spacing is off, check your Java code settings to be 4 spaces

peter-budo commented on app/src/com/ninetwozero/battlechat/activities/LoginActivity.java:
```diff
@@ -218,25 +219,39 @@ private void showProgress(final boolean show) {
 				connection = connection.data(
-					"email", params[0], 
-					"password", params[1],
+					"email", mSuppliedEmail,
```
Ugly m prefixes

peter-budo commented on app/src/com/ninetwozero/battlechat/activities/LoginActivity.java:
```diff
@@ -276,8 +292,12 @@ private boolean hasLoggedin(Connection.Response response) throws Exception {
-					new User(parser.getUserId(), parser.getUsername()),
-					CookieFactory.build(BattleChat.COOKIE_NAME, response.cookie(BattleChat.COOKIE_NAME)),
+					new User(
+                        parser.getUserId(),
```
Spacing is off

peter-budo commented on app/src/com/ninetwozero/battlechat/http/BattleChatClient.java:
```diff
@@ -83,7 +82,7 @@ public static JSONObject post(final String url, final int headerId, final NameVa
 
 	public static JSONObject getJsonObjectFromHttpResponse(HttpResponse response) throws Exception {
-        String message = "";
+        String message = "No message received.";
```
Extract to resources as string

peter-budo commented on core/src/main/java/com/ninetwozero/battlechat/datatypes/User.java:
```diff
@@ -66,14 +66,14 @@ public boolean isOffline() {
 				return "PLAYING";
 			default:
-				return "UNKNOWN";
+				return "ONLINE"; // 1 <= x < 264 = playing
```
Do you need that comment there?

peter-budo commented on core/src/test/java/com/ninetwozero/battlechat/http/LoginHtmlParserTest.java:
```diff
@@ -14,26 +17,33 @@
+        }
+
+        try {
+            stream.close();
```
I prefer to enclose it in if statement so if stream null no need to run close

```
if(stream != null){
    stream.close();
}
```

peter-budo commented on core/src/test/java/com/ninetwozero/battlechat/http/LoginHtmlParserTest.java:
```diff
@@ -44,7 +54,6 @@ public void testDetectmLoginError() {
 	@Test
 	public void testGetUserId() {
-    //    Assert.assertEquals(mLoggedIn.getUserId(), Long.parseLong("2832658801548551060"));
         Assert.assertEquals(mLoggedIn.getUserId(), 0);
```
import Assert as static import :wink:

peter-budo commented:
Also you may want to replace 1.0-SNAPSHOT with 1.0.5-SNAPSHOT

peter-budo committed:
Initial fixes

peter-budo committed:
Creating Lint violation to test CI game update

peter-budo committed:
- Fixed code indentation
- Removed unused libraries
- Removed unused TAGs in classes
- Updated some of dependent libraries
- Corrected pom version numbers

Commit:
Merge pull request #40 from peter-budo/pullRequestFix

Pull request fix

peter-budo committed:
Merging chat fix on dev branch with solving conflicts. However you need to have look on UserComparatorTest because it is failing. Looks like comparator is wrong.

Commit:
Merge pull request #43 from peter-budo/chat_fix_to_dev_merge

Merging chat-list-status-fix on dev branch

peter-budo commented:
Close it
